lib/api/controllers/adminController.js

```diff
@@ -398,4 +501,68 @@ function resetProfiles () {
     .then(() => this.internalEngine.createOrReplace('profiles', 'default', {policies: [{roleId: 'default'}]}));
 }
 
+function prepareSpecificationValidation(requestObject, kuzzle) {
+  var specifications = [];
+
+  return createSpecificationList(requestObject)
+    .then(list => {
+      specifications = list;
+      return validateSpecificationList(kuzzle, list);
+    })
+    .then(response => {
+      var responseObject;
+
+      if (response === true) {
+        return Promise.resolve({error: false, specifications: specifications});
+      }
+
+      responseObject = new ResponseObject(requestObject, response);
+      responseObject.error = 'Some errors with provided specifications.';
+      responseObject.status = 400;
+      return Promise.resolve({error: true, responseObject: responseObject});
```

Comment:
Or maybe reject the promise here so we don't have to check for an error in code using this method.